### PR TITLE
Compatibility with Laravel 10.x

### PR DESCRIPTION
**Pull request by colq2**
This PR bumps the dependencies so the package can be used with laravel 10.x

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -8,15 +8,17 @@ jobs:
     strategy:
       max-parallel: 15
       matrix:
-        laravel-version: ['^6.0', '^7.0', '^8.0', '^9.0']
-        php-versions: ['7.3', '7.4', '8.0']
+        laravel-version: ['^6.0', '^7.0', '^8.0', '^9.0', '^10.0']
```
Comment (atymic) [inline .github/workflows/php.yml:11]: Drop everything below 8 please

File: .github/workflows/php.yml [hunk 2]
```diff
@@ -8,15 +8,17 @@ jobs:
...
+        php-versions: ['7.3', '7.4', '8.0', '8.1']
```
Comment (atymic) [inline .github/workflows/php.yml:12]:
```suggestion
        php-versions: ['8.0', '8.1', '8.2']
```

File: .github/workflows/php.yml [hunk 3]
```diff
@@ -8,15 +8,17 @@ jobs:
...
         exclude:
           - laravel-version: ^6.0
             php-version: 8.0
           - laravel-version: ^7.0
             php-version: 8.0
           - laravel-version: ^9.0
             php-version: 7.4
+          - laravel-version: ^10.0
```
Comment (atymic) [inline .github/workflows/php.yml:20]: No need

File: .github/workflows/php.yml [hunk 4]
```diff
@@ -8,15 +8,17 @@ jobs:
     strategy:
       max-parallel: 15
       matrix:
-        laravel-version: ['^6.0', '^7.0', '^8.0', '^9.0']
-        php-versions: ['7.3', '7.4', '8.0']
+        laravel-version: ['^6.0', '^7.0', '^8.0', '^9.0', '^10.0']
+        php-versions: ['7.3', '7.4', '8.0', '8.1']
         exclude:
           - laravel-version: ^6.0
```
Comment (atymic) [inline .github/workflows/php.yml:14]: Delete 6 and 7 fom here

File: composer.json [hunk 5]
```diff
@@ -16,13 +16,13 @@
         }
     ],
     "require": {
-        "php": ">=7.4",
-        "illuminate/support": "~6.0 || ~7.0 || ~8.0 || ^9.0"
+        "php": ">8.1",
```
Comment (atymic) [inline composer.json:19]: Lets keep > 8.0 support please

File: composer.json [hunk 6]
```diff
@@ -16,13 +16,13 @@
...
+        "illuminate/support": "^10.0"
```
Comment (atymic) [inline composer.json:20]: Same here, 8+ for laravel and other deps

**Review by atymic — CHANGES_REQUESTED** [review]

**Comment by colq2** [comment]
@atymic thank you for reviewing this! I addressed your review. Just saw that there is another PR from laravel-shift.

**Comment by colq2** [comment]
@atymic is there anything else that is holding this back? I am happy to help

**Review by atymic — APPROVED** [review]

**Pull request merged**